lib/streamlit/elements/arrow.py

```diff
@@ -673,45 +688,60 @@ def table(self, data: Data = None) -> DeltaGenerator:
             .. |st.markdown| replace:: ``st.markdown``
             .. _st.markdown: https://docs.streamlit.io/develop/api-reference/text/st.markdown
 
+        border : bool or "horizontal"
+            Whether to show borders around the table and between cells. This can be one
+            of the following:
+
+            - ``True``: Show borders around the table and between cells (default)
+            - ``False``: Show no borders
+            - ``"horizontal"``: Show only horizontal borders between rows
+
         Examples
         --------
-        **Example 1: Display a simple dataframe as a static table**
+        **Example 1: Display a confusion matrix**
```

Comment:
Note that I updated the two examples here to show some more real-world use cases for `st.table`. I think this is important so we differentiate it from `st.dataframe`. @sfc-gh-dmatthews your call whether we should edit something here. We also need to update the demo apps. 

I'd also love to add a 3rd example with `border=None` that shows you can add a borderless table for layout purposes (e.g. to show some metric names on the left and values on the right side), but I think we first need `hide_index` and `hide_header` to make that look good (and ideally hide both by default if there's no data passed), so I left it out for now.

Reply:
This is what the 2 examples look like:

![Screenshot 2025-06-28 at 01 30 27](https://github.com/user-attachments/assets/6131854a-e9ef-4919-936e-b455cd979ab6)